modules/instance_template/main.tf

```diff
@@ -99,6 +99,13 @@ resource "google_compute_instance_template" "tpl" {
     network            = var.network
     subnetwork         = var.subnetwork
     subnetwork_project = var.subnetwork_project
+    dynamic "access_config" {
+      for_each = var.public_ip ? [var.access_config] : []
+      content {
+        nat_ip       = lookup(access_config.value, "nat_ip", null)
```

Comment:
```suggestion
        nat_ip                 = access_config.value.nat_ip
        public_ptr_domain_name = access_config.value.public_ptr_domain_name
```

Reply:
yes, I also tried using `public_ptr_domain_name` according to the documentation https://www.terraform.io/docs/providers/google/r/compute_instance.html#access_config, but it throws
```
       Error: Unsupported argument

         on ../../../../modules/instance_template/main.tf line 107, in resource "google_compute_instance_template" "tpl":
        107:         public_ptr_domain_name = access_config.value.public_ptr_domain_name

       An argument named "public_ptr_domain_name" is not expected here.
```
that's why I dropped the argument

Reply:
Oh, my mistake. That's only support on instances, not instance templates.